Woof/Woof/ViewModel/Owner/OwnerProfileViewModel.swift

```diff
@@ -26,48 +26,52 @@ class OwnerProfileViewModel: ObservableObject {
     /// Indicates whether the logout confirmation was obtained.
     @Published var isLogoutConfirmed = false
 
+    /// Detailed error information for the user.
+    @Published var errorMessage: String?
+
+    /// Indicates if the mandatory fields are empty.
+    var mandatoryFieldsAreEmpty: Bool {
+        name.isEmpty || phone.isEmpty || city.isEmpty || address.isEmpty
+    }
+
     /**
      Initializes an instance of the `OwnerProfileViewModel` class.
      */
     init() {
-        currentOwner = getCurrentOwner()
-
-        name = currentOwner.name
-        surname = currentOwner.surname
-        phone = currentOwner.phone
-        city = currentOwner.city
-        address = currentOwner.address
-        avatarURL = currentOwner.avatarURL
+        setUpFields()
     }
 
     /// Requests model layer to save modified data.
     func save() {
-        currentOwner.name = name
-        currentOwner.surname = surname
-        currentOwner.phone = phone
-        currentOwner.city = city
-        currentOwner.address = address
-        currentOwner.avatarURL = avatarURL
-
-        guard let data = try? JSONEncoder().encode(currentOwner) else { return }
-
-        KeyValueStorage(KeyValueStorage.Name.currentOwner)
-            .save(data, for: KeyValueStorage.Key.currentOwner)
+        var newOwner = currentOwner ?? .init()
+        newOwner.name = name
+        newOwner.surname = surname
+        newOwner.phone = phone
+        newOwner.city = city
+        newOwner.address = address
+        newOwner.avatarURL = avatarURL
+
+        if let data = try? JSONEncoder().encode(newOwner),
+           KeyValueStorage(KeyValueStorage.Name.currentOwner)
+           .save(data, for: KeyValueStorage.Key.currentOwner) {
+        } else {
+            errorMessage = AppError.saveLocallyFailed.errorDescription
+        }
     }
 
     /**
      Returns the owner information that will be displayed in view.
 
-        - Returns: The owner instance from model layer or new instance of `Owner`,
+     - Returns: The owner instance from model layer or new instance of `Owner`,
      if loading from model layer is failed.
      */
-    func getCurrentOwner() -> Owner {
-        loadOwnerFromStorage() ?? Owner()
+    func getCurrentOwner() -> Owner? {
+        loadOwnerFromStorage()
```

Comment:
Could you please update the doc comments as we returned an optional value.

Reply:
Fixed